main.py

```diff
@@ -184,14 +190,24 @@ def upgrade_dockerfile(self, repo):
         try:
             existing_file = repo.get_contents(filename)
         except GithubException:
+            print('No Dockerfile file found')
             return  # doesn't have a Dockerfile
 
         sha = existing_file.sha
         existing_content = base64.b64decode(existing_file.content) \
             .decode('UTF-8').strip()
 
         content = existing_content.split('\n')
-        content[0] = f'FROM ruby:{self.new_version}'
+        old = f'ruby:{self.old_version}'
+        new = f'ruby:{self.new_version}'
+
+        for line in content:
```

Comment:
I wonder if instead of splitting, looping, then joining, this section could be simply written as something like:

```python
old = f'ruby:{self.old_version}'
new = f'ruby:{self.new_version}'
if re.search(old, existing_content):
    content = re.sub(old, new, existing_content)
else:
    return
```

Reply:
That sounds like a better approach, I was just trying not to get too excited with refactoring the whole thing. Will try that though, thanks!